File: trabalho-ps/src/main/java/com/afm/trabalho_ps/dto/RelatorioVendasDTO.java
package com.afm.trabalho_ps.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public class RelatorioVendasDTO {
    public static class VendaResumo {
        public Long id;
        public LocalDate data;
        public String cliente;
        public BigDecimal valorTotal;
        public List<ItemResumo> itens;
    }
    public static class ItemResumo {
        public String produto;
        public int quantidade;
    }

    public List<VendaResumo> vendas;
    public BigDecimal totalVendido;
    public int totalItensVendidos;
    public String produtoMaisVendido;
    public int quantidadeMaisVendida;
}

File: front-end/src/interfaces/RelatorioVendasDTO.ts
export interface RelatorioVendasDTO {
  totalVendido: number;
  totalItensVendidos: number;
  produtoMaisVendido: string;
  quantidadeMaisVendida: number;
  vendas: VendaResumo[];
}

export interface VendaResumo {
  id: number;
  data: string;
  cliente: string;
  valorTotal: number;
  itens: ItemResumo[];
}

export interface ItemResumo {
  produto: string;
  quantidade: number;
}
